src/readField.js

```diff
@@ -115,7 +115,7 @@ const readField = (state, fieldName, pathToHere = '', fields, syncErrors, asyncV
     const initialFormValue = read(`${name}.initial`, form);
     const initialValue = initialFormValue || read(name, initialValues);
     field.name = name;
-    field.checked = initialValue === true || undefined;
+    field.checked = String(initialValue).toLowerCase() === 'true' || undefined;
```

Comment:
I don't like this condition... A more descriptive test would have been better in my opinion, something like:

```js
initialValue === true || initialValue === "true" || undefined
```